File: Script4Test/cs/Script4Test.cs
using System.Collections.Generic;

// Только тело метода становится блоками; using и шапка класса переносятся
// как есть. Внутри — письменное подмножество NSG. Дженерики — в текстовых
// слотах типа, поэтому переводятся блоком, а не сырым фрагментом.
public class Script4Test
{
    public int Crunch(int n)
    {
        List<int> sieve = new List<int>();
        Dictionary<string, int> tally = new Dictionary<string, int>();
        int primes = 0;
        int i = 0;
        while (i <= n)
        {
            sieve.Add(0);
            i++;
        }
        i = 2;
        while (i <= n)
        {
            if (sieve[i] == 0)
            {
                primes += 1;
                tally["p" + primes] = i;
                int j = i * i;
                while (j <= n)
                {
                    sieve[j] = 1;
                    j += i;
                }
            }
            i++;
        }
        int mask = (1 << 8) - 1;
        int folded = (primes & mask) ^ (primes >> 3);
        int bonus = folded > 0 ? folded : -1;
        return primes + bonus + tally.Count;
    }
}

File: Script4Test/go/script4test.go
package script4test

// Только тело функции становится блоками; package и шапка переносятся как есть.
// Внутри — письменное подмножество NSG.
func Crunch(n int) int {
	primes := 0
	i := 2
	for i <= n {
		isPrime := 1
		j := 2
		for j*j <= i {
			if i%j == 0 {
				isPrime = 0
			}
			j += 1
		}
		primes += isPrime
		i += 1
	}
	mask := (1 << 8) - 1
	folded := (primes & mask) ^ (primes >> 3)
	bonus := -1
	if folded > 0 {
		bonus = folded
	}
	return primes + bonus
}
